lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2228,6 +2228,22 @@ ModuleDecl *ClangImporter::Implementation::finishLoadingClangModule(
     (void) namelookup::getAllImports(result);
   }
 
+  // Register '.h' inputs of each Clang module dependency with
+  // the dependency tracker. In implicit builds such dependencies are registered
+  // during the on-demand construction of Clang module. In Explicit Module
+  // Builds, since we load pre-built PCMs directly, we do not get to do so. So
+  // instead, manually register all `.h` inputs of Clang module dependnecies.
+  if (SwiftDependencyTracker &&
+      !Instance->getInvocation().getLangOpts().ImplicitModules) {
+    auto *moduleFile = Instance->getASTReader()->getModuleManager().lookup(
+        clangModule->getASTFile());
+    Instance->getASTReader()->visitInputFileInfos(
+        *moduleFile, /*IncludeSystem=*/true,
+        [&](const clang::serialization::InputFileInfo &IFI, bool isSystem) {
+          SwiftDependencyTracker->addDependency(IFI.Filename, isSystem);
+        });
+  }
```

Comment:
nitpick: not sure if we need some logics like `needSystemDependencies()` for skipping system headers? or is it handled already.

Reply:
I believe that's already handled under the hood in the Dependency Tracker, hence why we are passing in `isSystem`. But I'll double-check.

Reply:
Yeah, looks like it is. Without `-track-system-dependencies` this will not add any of the system framework headers. 👍🏼